### PR TITLE
enumerate required org webhooks

### PR DESCRIPTION
**Pull request by RaphaelVogel**

### CODE REVIEW

File: concourse/util.py [hunk 1]
```diff
@@ -173,3 +177,51 @@ def url_filter(url):
...
+            github_api = _create_github_api_object(
+                github_cfg=cfg_factory.github(github_cfg_name),
+            )
+
+            webhook_url = _build_webhook_url(
+                ingress_host_url=whd_deployment_cfg.ingress_host(),
+                whd_cfg_name=whd_cfg_name,
+            )
+
+            webhook_syncer = github.webhook.GithubWebHookSyncer(github_api)
```
Comment (ccwienk) [inline concourse/util.py]: mh?

File: concourse/util.py [hunk 2]
```diff
@@ -173,3 +177,51 @@ def url_filter(url):
...
+def _enumerate_github_org_configs(job_mapping_set: JobMappingSet,):
+    '''Returns tupes of github org names and github config names'''
+    for job_mapping in job_mapping_set.job_mappings().items():
+        github_org_configs = job_mapping.github_organizations()
+
+        for github_org_config in github_org_configs:
+            yield (github_org_config.org_name(), github_org_config.github_cfg_name)
+
+
+def _build_webhook_url(ingress_host_url: str, whd_cfg_name: str):
```
Comment (ccwienk) [inline concourse/util.py]: yeah, well... why not inline this call?

File: concourse/util.py [hunk 3]
```diff
@@ -173,3 +177,51 @@ def url_filter(url):
...
+                ingress_host_url=whd_deployment_cfg.ingress_host(),
+                whd_cfg_name=whd_cfg_name,
+            )
+
+            webhook_syncer = github.webhook.GithubWebHookSyncer(github_api)
+            # TODO: now sync the webhooks
+
+
+def _enumerate_github_org_configs(job_mapping_set: JobMappingSet,):
+    '''Returns tupes of github org names and github config names'''
```
Comment (ccwienk) [inline concourse/util.py]: tupLes

File: concourse/util.py [hunk 4]
```diff
@@ -173,3 +177,51 @@ def url_filter(url):
...
+    whd_cfg = cfg_factory.webhook_dispatcher(whd_cfg_name)
+
+    concourse_cfg_names = whd_cfg.concourse_config_names()
+    concourse_cfgs = map(cfg_factory.concourse, concourse_cfg_names)
+
+    for concourse_cfg in concourse_cfgs:
+        job_mapping_set = cfg_factory.job_mapping(concourse_cfg.job_mapping_cfg_name())
+
+        for github_orgname, github_cfg_name in _enumerate_github_org_configs(job_mapping_set):
+            github_api = _create_github_api_object(
```
Comment (ccwienk) [inline concourse/util.py]: github_api is not used (yet?). so why create it?

File: concourse/util.py [hunk 5]
```diff
@@ -173,3 +177,51 @@ def url_filter(url):
...
+                github_cfg=cfg_factory.github(github_cfg_name),
+            )
+
+            webhook_url = _build_webhook_url(
+                ingress_host_url=whd_deployment_cfg.ingress_host(),
+                whd_cfg_name=whd_cfg_name,
+            )
```
Comment (ccwienk) [inline concourse/util.py]: so actually, my suggestions would be:
- add a `yield (github_api, github_org_name, webhook_url)` here
- rename this function (something like "enumerate_required_org_webhooks")
- call this function from a (then new) function "sync(_org?)_webhooks; iterate over the returned three-tuples in said function

Comment (RaphaelVogel) [reply concourse/util.py]: Done

**Pull request merged**